shinhye/app.js

```diff
@@ -47,7 +48,135 @@ app.post("/users/signup", async (req, res) => {
     [username, firstName, lastName, age, email, password]
   );
 
-  res.status(201).json({ message: "Successfully created!" });
+  return res.status(201).json({ message: "Successfully created!" });
+});
+
+// add a post
+app.post("/posts", async (req, res) => {
+  const { userID, userProfileImage, postImageURL, postContent } = req.body;
+
+  await AppDataSource.query(
+    `INSERT INTO posts(
+      user_id, 
+      user_profile_image,
+      post_image_url,
+      post_content
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [userID, userProfileImage, postImageURL, postContent]
+  );
+
+  return res.status(201).json({ message: "Successfully created!" });
+});
+
+// view all posts
+app.get("/posts", async (req, res) => {
+  const postData = await AppDataSource.query(
+    `
+    SELECT
+    user_id as userID,
+    user_profile_image as userProfileImage,
+    id as postId,
+    post_image_url as postImageUrl,
+    post_content as postContent
+    FROM posts;
+    `
+  );
+
+  return res.status(200).send({ data: postData });
+});
+
+// view user's post
+app.get("/posts/:userID", async (req, res) => {
+  const { userID } = req.params;
+
+  const userPosts = await AppDataSource.query(
+    `SELECT
+      p.user_id as userID,
+      p.user_profile_image as userProfileImage,
+      JSON_ARRAYAGG(
+        JSON_OBJECT(
+        'postingId', p.id,
+        'postImageUrl', p.post_image_url,
+        'postContent', p.post_content))
+      AS postings
+    FROM posts p
+    WHERE p.user_id = ?
+    GROUP BY user_id, user_profile_image;
+    `,
+    [userID]
+  );
+  res.status(200).json({ data: userPosts });
+});
+
+// update user's post
+app.patch("/posts/:userID/:postID", async (req, res) => {
+  const { userID, postID } = req.params;
+  const { postContent } = req.body;
+
+  await AppDataSource.query(
+    `
+    UPDATE posts
+    SET
+    post_content = ?
+    WHERE user_id = ?
+    AND id = ?;
+    `,
+    [postContent, userID, postID]
+  );
+
+  const updatedPost = await AppDataSource.query(
+    `
+    SELECT
+    users.id as userID,
+    users.username,
+    posts.id as postID,
+    posts.post_content as postContent
+    FROM users
+    INNER JOIN posts ON posts.user_id = users.id
+    WHERE posts.user_id = ? 
+    AND posts.id= ?;
+    `,
+    [userID, postID]
+  );
+
+  return res.status(200).json({ data: updatedPost });
+});
+
+// delete post
+app.delete("/posts/:postID", async (req, res) => {
```

Comment:
마이너하지만 path-parameter 네이밍 수정해주세요!

```suggestion
app.delete("/posts/:postId", async (req, res) => {
```

Reply:
넵 반영했습니다!